strapi/strapi
# [Feature] Populate fragments for polymorphic relations

### What does it do?

Add the possibility to fragment the populate query of polymorphic relations using the `populate.on` property.
It's possible to use this syntax for dynamic zones and media to add more granularity & precision to the populate's queries. 

### Why is it needed?

Up until now, it was not possible to differentiate multiple entities attached to a polymorphic relation & apply different rules to them based on their type.
It was particularly an issue regarding dynamic zone population as it was thus impossible to apply different filters to different kinds of component structures.

### How to test it?

- Run the `examples/getstarted` application
- Create a restaurant entity and add multiple (different) components to the dynamic zone value (`dz` attribute) 
- Go to your browser and use a query similar to
`http://localhost:1337/api/restaurants?populate[dz][on][default.dish]=true`
You can also have variant such as
`http://localhost:1337/api/restaurants?populate[dz][on][default.dish][fields]=name`
- It should only populate the components that are defined in the `on` property.

### Related issue(s)/PR(s)

fix https://github.com/strapi/strapi/issues/13220
